src/components/common/CarouselComponent.jsx

```diff
@@ -0,0 +1,127 @@
+import { useRef, useState, useEffect, Children, cloneElement } from 'react';
+import styled from 'styled-components';
+
+const CarouselContainer = styled.div`
+  width: 100%;
+  overflow-x: auto;
+  overflow-y: hidden;
+  display: flex;
+  gap: ${({ $gap }) => $gap};
+
+  /* 스크롤바 숨김 */
+  scrollbar-width: none;
+  -ms-overflow-style: none;
+  &::-webkit-scrollbar {
+    display: none;
+  }
+
+  /* 스크롤 스냅 */
+  scroll-snap-type: x mandatory;
+  scroll-behavior: smooth;
+  -webkit-overflow-scrolling: touch;
+
+  cursor: grab;
+  &:active {
+    cursor: grabbing;
+  }
+  -webkit-user-select: none;
+  -moz-user-select: none;
+  user-select: none;
+
+  touch-action: pan-y;
+
+  > * {
+    flex: 0 0 100%; // 각 아이템이 컨테이너 너비를 100% 차지하도록 수정
+    scroll-snap-align: start;
+    scroll-snap-stop: always;
+  }
+`;
+
+const Carousel = ({ children, gap }) => {
+  const carouselRef = useRef(null);
+  const [firstX, setFirstX] = useState(0);
+  const [lastX, setLastX] = useState(0);
+
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const observerRef = useRef(null);
+
+  useEffect(() => {
+    if (!carouselRef.current) return;
+    if (observerRef.current) {
+      observerRef.current.disconnect();
+    }
+
+    observerRef.current = new IntersectionObserver(
+      (entries) => {
+        if (entries.length > 0) {
+          const index = parseInt(entries[0].target.dataset.index);
+          setCurrentIndex(index);
+        }
+      },
+      {
+        root: carouselRef.current,
+        threshold: 1.0,
+        rootMargin: '0px',
+      },
+    );
+
+    const domChildren = carouselRef.current.children;
+    Array.from(domChildren).forEach((child) => {
+      observerRef.current.observe(child);
+    });
+
+    return () => {
+      if (observerRef.current) {
+        observerRef.current.disconnect();
+      }
+    };
+  }, []);
```

Comment:
❗ MultiItemCarousel 상황에서는 인덱스가 필요하지 않으므로, setIndex를 옵셔널 props로 생각하고, 호출하는 쪽에서 setIndex를 넘기지 않으면 인덱스를 추적하지 않기로 해요!
```suggestion
if(setIndex) {
  const observerRef = useRef(null);

  useEffect(() => {
    if (!carouselRef.current) return;
    if (observerRef.current) {
      observerRef.current.disconnect();
    }

    observerRef.current = new IntersectionObserver(
      (entries) => {
        if (entries.length > 0) {
          const index = parseInt(entries[0].target.dataset.index);
          setCurrentIndex(index);
        }
      },
      {
        root: carouselRef.current,
        threshold: 1.0,
        rootMargin: '0px',
      },
    );

    const domChildren = carouselRef.current.children;
    Array.from(domChildren).forEach((child) => {
      observerRef.current.observe(child);
    });

    return () => {
      if (observerRef.current) {
        observerRef.current.disconnect();
      }
    };
  }, []);
}
```